src/services/user/RegistrarAcesso.ts

```diff
@@ -0,0 +1,27 @@
+import { IUser } from '../../entities/User'
+import UsersRepository, {
+  IUsersRepository
+} from '../../repositories/UsersRepository'
+import {
+  agendaNotificacaoTresDias,
+  cancelaNotificacoesAgendadas
+} from '../../utils/notificacoes'
+
+interface IRegistrarAccesso {
+  call(user: IUser): Promise<void>
+}
+
+export default class RegistrarAccesso implements IRegistrarAccesso {
+  private usersRepository: IUsersRepository
+
+  constructor() {
+    this.usersRepository = new UsersRepository()
+  }
+
+  async call(user: IUser): Promise<void> {
+    if (this.usersRepository.updateAccessFlags(user)) {
```

Comment:
apesar dessa função retornar true quando atualiza e false quando não atualiza, essa atualização na vdd é referente ao teste se está ou não no mesmo dia, olhando o if e depois a função eu achei que não ficou mto óbvio.

pra ficar mais explícito eu recomendaria algo assim:
```suggestion
    const accessFlagsUpdated = this.usersRepository.updateAccessFlags(user)
    if (accessFlagsUpdated) {
```